src/app/domain/osReceive/services/OsReceiveData.ts

```diff
@@ -42,13 +45,19 @@ const mapFilesToUploadToArray = (
   filesMap: Map<string, ReceivedFile>
 ): OsReceiveFile[] => {
   return Array.from(filesMap.values()).map(file => ({
-    name: file.fileName,
+    name: decodeURI(file.fileName),
     file: {
       ...file,
-      fromFlagship: true
+      fromFlagship: true,
+      filePath: decodeURI(file.filePath),
+      fileName: decodeURI(file.fileName),
+      mimeType: getMimeType({
+        name: decodeURI(file.fileName),
+        mimeType: Platform.OS === 'android' ? file.mimeType : undefined // mimeType is not reliable on iOS
+      } as Media)
     },
     status: OsReceiveFileStatus.toUpload,
-    type: file.mimeType
+    type: 'application/pdf'
```

Comment:
what about jpg/mov or else? 🤔

Reply:
good catch wasn't supposed to be there at all. I guess the unit test failed at detecting it, I'm going to update it too

Reply:
Did we check the iOS code to see why we can't rely on the mimeType? It sounds weird 🤔

Reply:
well in iOS the mimetype is just ".pdf" for pdf for instance. The stack does not recognize it and the file has no icon. (not really breaking in itself). On Android the mimeType has the expected web format